### PR TITLE
[!!!][TASK] Remove deprecated plugin and sub types functionality

### PR DESCRIPTION
**Pull request by linawolf**
Resolves: https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/1152
Releases: main

### CODE REVIEW

File: Documentation/ApiOverview/ContentElements/ContentElementsWizard.rst [hunk 1]
```diff
@@ -74,19 +74,15 @@ of class :php:`\TYPO3\CMS\Core\Utility\ExtensionManagementUtility`.
...
 content element representing the plugin.
 
-When you are using `CType` for parameter `$type` the content
-element is added to the select item list of column `CType` in table `tt_content`.
+..  versionchanged:: 14.0
 
-..  deprecated:: 13.4
-    Using the default `list_type` for the parameter is deprecated. All content
-    elements and plugins should be added with string `CType` for parameter `$type`.
+    The methods second and third parameter have been dropped. This method can
```
Comment (garvinhicking) [inline Documentation/ApiOverview/ContentElements/ContentElementsWizard.rst:79]:
```suggestion
    The method's second and third parameter have been dropped. This method can
```

File: Documentation/ApiOverview/ContentElements/Index.rst [hunk 2]
```diff
@@ -185,9 +185,10 @@ To register such a plugin as content element you can use function
 ..  literalinclude:: _Plugins/_tt_content_plugin.php
     :caption: EXT:my_extension/Configuration/TCA/Overrides/tt_content.php
 
-..  deprecated:: 13.4
-    Setting the second parameter to `list_type`
-    is deprecated. See :ref:`plugins-list_type-migration`.
+..  versionchanged:: 14.0
+
+    The methods second and third parameter have been dropped. This method can
```
Comment (garvinhicking) [inline Documentation/ApiOverview/ContentElements/Index.rst:190]:
```suggestion
    The method's second and third parameter have been dropped. This method can
```

**Comment by Alagts** [comment]
Local rendering revealed one warning: 
[2024-11-14T10:54:35.360449+00:00] app.ERROR: Error while processing "literalinclude" directive in "ExtensionArchitecture/BestPractises/NamingConventions": Include "_snippets/_tt_content_plugin_htmlparser.php" (/ExtensionArchitecture/BestPractises/_snippets/_tt_content_plugin_htmlparser.php) does not exist or is not readable. {"rst-file":"ExtensionArchitecture/BestPra...

The directive 

```
..  literalinclude:: _snippets/_tt_content_plugin_htmlparser.php  
    :caption: EXT:examples/Configuration/TCA/Overrides/tt_content_plugin_htmlparser.php
```

has therefore no effect in Documentation/ExtensionArchitecture/BestPractises/NamingConventions.rst. Just add the file "_tt_content_plugin_htmlparser.php" in /ExtensionArchitecture/BestPractises/_snippets/ then all would be fine to merge.

**Comment by linawolf** [comment]
Good catch, I wonder why the rendering pipeline did not warn about that. Resolved

**Pull request merged**